tools/qontract_cli.py

```diff
@@ -3770,5 +3783,40 @@ def remove(ctx, sso_client_vault_secret_path: str):
     )
 
 
+@root.group()
+@click.pass_context
+def external_resources(ctx):
+    """External resources commands"""
+
+
+@external_resources.command()
+@click.argument("provision-provider", required=True)
+@click.argument("provisioner", required=True)
+@click.argument("provider", required=True)
+@click.argument("identifier", required=True)
+@click.pass_context
+def get_input(
+    ctx, provision_provider: str, provisioner: str, provider: str, identifier: str
+):
+    namespaces = [ns for ns in get_namespaces() if ns.external_resources]
+    er_inventory = ExternalResourcesInventory(namespaces)
+
+    spec = er_inventory.get_inventory_spec(
+        provision_provider=provision_provider,
+        provisioner=provisioner,
+        provider=provider,
+        identifier=identifier,
+    )
+    vault_settings = get_app_interface_vault_settings()
+    secret_reader = create_secret_reader(use_vault=vault_settings.vault)
+    er_settings = get_settings()[0]
+    m_inventory = load_module_inventory(get_modules())
+    factories = setup_factories(er_settings, m_inventory, er_inventory, secret_reader)
+    f = factories.get_factory(spec.provision_provider)
+    resource = f.create_external_resource(spec)
+    f.validate_external_resource(resource)
+    print(json.dumps(resource.dict(exclude={"data": {FLAG_RESOURCE_MANAGED_BY_ERV2}})))
```

Comment:
any requirement to use `json.dumps` instead of directly `resource.json()`? And how to verify this command? [unit tests](https://github.com/app-sre/qontract-reconcile/blob/836f3bbb0a07648c2049d198a52dc4afaea0ada3/tools/test/test_qontract_cli.py) or some examples in doc?

Reply:
Changed to resource.json() and added an example in the docs.

Reply:
I have this error in output

```
reconcile.utils.exceptions.FetchResourceError: error fetching resource: Resource spec not found: Provider rds, Id: dashdotdb-stage
```

Any pre requirement to use this command?

Reply:
This resource is not yet managed by ErV2. It's just an example to show how the command works. If you want a working example: 

`qontract-cli --config=config.debug.toml external-resources get-input aws app-sre-stage rds erv2-tests` 

But this resource will be removed so it's not worth using it as an example.

This command will be used in the SOP to migrate resources along with all the required steps. It will make more sense once the full SOP is written.

Reply:
Is this for human usage or machine usage? Can use `json(indent=2` to pretty print result